src/settingsview.cpp

```diff
@@ -127,6 +128,7 @@ void SettingsView::clearMonitorDir()
     if (confirmDialog(gt("monitor-clear-dir-dialog-msg"), gt("monitor-clear-dir-dialog-title"))) {
         ui->monitorClear->hide();
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
+        KiwixApp::instance()->setMonitorDir("");
         KiwixApp::instance()->getLibrary()->setMonitorDirZims(QStringList());
```

Comment:
`KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);` and `KiwixApp::instance()->getLibrary()->setMonitorDirZims(QStringList());` can be moved into `KiwixApp::setMonitorDir()`.

Reply:
Do you mean the calls to those functions or full logic?

Reply:
I was asking if you mean to remove `SettingsManager::setMonitorDir` and `Library::setMonitorDirZims` from the app and use `KiwixApp::setMonitorDir` only OR just use them two functions in `KiwixApp::setMonitorDir`, but now I think you meant the latter